packages/task/src/browser/task-schema-updater.ts

```diff
@@ -74,6 +77,8 @@ export class TaskSchemaUpdater {
                 fileMatch: ['tasks.json'],
                 url: taskSchemaUri.toString()
             });
+        } finally {
+            this.onTaskSchemaChangedEmitter.fire(undefined);
```

Comment:
I wonder should not we rather add generic `onDidChangeSchema(uri => {}i)` event to `JsonSchemaStorage` since it's the primary source.

Reply:
@akosyakov 
I thought about event from `JsonSchemaStorage` and it was the first thing that I tried to use.
But I found that at starting app `registerSchema()` is invoke about 182 times and only a few times `notifyChanged()`.
I think it cause by using [debounce](https://github.com/eclipse-theia/theia/blob/0f16fe89b928e67ddea08b637a4cce393dea1d85/packages/core/src/browser/json-schema-store.ts#L41).  

So it's not clear for me how we should resolve this case.
Maybe I missed something?
Could you advise what solution will be better?
thank you!

Reply:
Ideally it would be good to add a new event, you don't need to use `notifyChanged`, but add a new event `onDidChangeSchema` which is not debounced for now and provide and uri.

then `onTaskSchemaChanged` can be fired like
```ts
jsonSchemaStore.onDidChangeSchema(uri => {
   if (uri === taskSchemaUri) {
      this.onTaskSchemaChangedEmitter.fire(undefined)
   }
});
```

Later we can optimize `onDidChangeSchema`.

btw https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#event_names

Reply:
thank you very much for your help!
I updated the PR, could you take a look when you have a chance.